tests/random.py

```diff
@@ -97,6 +97,24 @@ def bounds_check(t):
 prob_bounds = st.tuples(none_or_(prob), none_or_(prob)).filter(bounds_check)
 
 
+# settings
+def force_settings(name, value):
```

Comment:
Aha, now I see what you meant. What if we change the hbar attribute of settings just for the tests? Or the ImmutableSetting class has a test_mode: bool that allows for mutability?